File: core/src/main/java/com/lomekwi/cave/timeline/SegmentSelectedEvent.java
package com.lomekwi.cave.timeline;

import com.lomekwi.cave.app.selection.SelectableSelectedEvent;
import org.jspecify.annotations.Nullable;

public record SegmentSelectedEvent(@Nullable Segment segment, @Nullable Track track, int selectedCount)
    implements SelectableSelectedEvent<Segment> {
    @Override
    public @Nullable Segment selectable() { return segment; }
}

File: core/src/main/scala/com/lomekwi/cave/timeline/SegmentSelectedEvent.scala
package com.lomekwi.cave.timeline

import com.lomekwi.cave.app.selection.SelectableSelectedEvent

case class SegmentSelectedEvent(segment: Segment, track: Track, selectedCount: Int)
    extends SelectableSelectedEvent[Segment] {
  override def selectable(): Segment = segment
}
